fili-core/src/test/groovy/com/yahoo/bard/webservice/async/AsyncDruidSendsErrorSpec.groovy

```diff
@@ -0,0 +1,90 @@
+package com.yahoo.bard.webservice.async
+
+import static com.yahoo.bard.webservice.async.jobs.jobrows.DefaultJobStatus.FAILURE
+import static com.yahoo.bard.webservice.async.jobs.jobrows.DefaultJobStatus.PENDING
+
+import com.yahoo.bard.webservice.util.GroovyTestUtils
+
+/**
+ * Verifies that the job status is updated to error, and results links return the error message when Druid experiences
+ * an error.
+ */
+class AsyncDruidSendsErrorSpec extends AsyncFunctionalSpec {
+
+    static final String QUERY =
+            "http://localhost:9998/data/shapes/day?dateTime=2016-08-30%2F2016-08-31&metrics=height&asyncAfter=0"
+
+    static final String ERROR_MESSAGE = """{
+                        "status" : 500,
+                        "statusName" : "Internal Server Error",
+                        "reason" : "All the things have broken.",
+                        "description" : "All the things have broken.",
+                        "druidQuery" : null
+                    }"""
+
+    @Override
+    Map<String, Closure<String>> getResultsToTargetFunctions() {
+        return [
+                data: { "data/shapes/day" },
+                //By querying the syncResults link first, we wait until the results are ready, thanks to the
+                //BroadcastChannel. So we have no race condition between the data request updating the job ticket, and
+                //the query for the job metadata
+                syncResults: { AsyncTestUtils.extractTargetFromField(it.data.readEntity(String), "syncResults") },
+                results: { AsyncTestUtils.extractTargetFromField(it.data.readEntity(String), "results") },
+                jobs: { AsyncTestUtils.buildTicketLookup(it.data.readEntity(String)) }
+        ]
+    }
+
+    @Override
+    Map<String, Closure<Void>> getResultAssertions() {
+        return [
+                data: {
```

Comment:
I feel like having comments in here indicating the scenario and what the assertion is expecting would help aid in understanding both what behavior is being tested for, as well as what the queries are for which the behavior is expected.